### PR TITLE
tsb: use external clients

### PR DESCRIPTION
**Pull request by jim-minter**
https://trello.com/c/LclZtOcc/1382-5-switch-template-service-broker-to-use-external-clients

### CODE REVIEW

File: pkg/templateservicebroker/util/util.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package util
```
Comment (bparees) [inline pkg/templateservicebroker/util/util.go]: i'm not convinced this is the right location for this helper.  it doesn't seem TSB specific.  Perhaps it even belongs under pkg/authorization/util/external or something, which would put it closer to where it came from?

@deads2k wdyt?

Comment (deads2k) [reply pkg/templateservicebroker/util/util.go]: > i'm not convinced this is the right location for this helper. it doesn't seem TSB specific. Perhaps it even belongs under pkg/authorization/util/external or something, which would put it closer to where it came from?
> 
> @deads2k wdyt?

After we can get a 3.8 rebase and make openshift/client-go authoritative, we could compare it for similar code and see if we have something generic to add.  Until then, I'd suggest trying to limit TSB's exposure to openshift/origin by whatever means necessary

**Comment by deads2k** [comment]
@jim-minter real unit test failure.

**Comment by jim-minter** [comment]
pushed to fix unit test failure, required rebase

File: pkg/template/generated/listers/template/v1/template_expansion.go [hunk 2]
```diff
@@ -2,26 +2,27 @@ package v1
 
 import (
 	templateapi "github.com/openshift/origin/pkg/template/apis/template"
```
Comment (deads2k) [inline pkg/template/generated/listers/template/v1/template_expansion.go]: This dependency is illegal.  I'm try to push through, but some debt is going to fall out.

**Pull request merged**